### DISCUSSION TITLE
I have CUDA installed yet can't select GPU as an option in the webui

### DISCUSSION THREAD

**Discussion by Rettroix** [absadiki/subsai#36] [Q&A] [answered]
I have CUDA installed yet can't select GPU as an option in the webui. Not sure what I could be doing wrong. Anyone encountered this? The only options available are none and CPU.

**Comment by absadiki** [comment]
@Rettroix,
If you cannot select the GPU then `Pytorch` does not recognize that you have a compatible `GPU`. 
There are so many things you need to investigate to check why it is the case.

First, check it manually by running:

```Python
>>> import torch
>>> torch.cuda.is_available()
True  # if cuda is available you will get True otherwise you will get False
```
if it returns `False` and you believe you have a compatible GPU then something is wrong with your installation, you can check this [stackoverflow thread](https://stackoverflow.com/questions/60987997/why-torch-cuda-is-available-returns-false-even-after-installing-pytorch-with) and go through the solutions there.
Hope this will help!

**Reply by Rettroix** [reply]
Thank you for your reply. I have checked manually as you sugested and torch.cude.is_available() is appearing false! I will investigate this further and get back when I have an answer as I am certain my GPU is compatible.

**Reply by Rettroix** [reply, marked as answer]
So I've managed to fix this and the solution is rather simple! All I had to do as uninstall torch with this command: 
`pip uninstall torch torchvision functorch tinycudann` 

and then resintall it with CUDA with this command:
`pip install torch==2.0.1+cu118 torchvision==0.15.2+cu118 --extra-index-url https://download.pytorch.org/whl/cu118` 

These commands were the ones found on this site: [https://docs.nerf.studio/en/latest/quickstart/installation](https://docs.nerf.studio/en/latest/quickstart/installation.html)

**Reply by absadiki** [reply]
@Rettroix, That's great! 
Thank you very much for posting the solution. It might help someone facing the same issue.
